src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -128,90 +142,104 @@ public void RelayWorker()
 
-                        break;
+                        if (this.blocksToAnnounce.TryDequeue(out ChainedBlock blockToBroadcast))
+                        {
+                            this.logger.LogTrace("Block '{0}' moved from the announce queue to broadcast list.", block);
+                            broadcastItems.Add(blockToBroadcast);
+                        }
+                        else this.logger.LogTrace("Unable to removing block '{0}' from the announce queue.", block);
                     }
 
-                    if (this.blocksToAnnounce.TryDequeue(out ChainedBlock blockToBroadcast))
+                    if (!broadcastItems.Any())
                     {
-                        this.logger.LogTrace("Block '{0}' moved from the announce queue to broadcast list.", block);
-                        broadcastItems.Add(blockToBroadcast);
+                        this.logger.LogTrace("(-)[NO_BROADCAST_ITEMS]");
+                        continue;
                     }
-                    else this.logger.LogTrace("Unable to removing block '{0}' from the announce queue.", block);
-                }
 
-                if (!broadcastItems.Any())
-                {
-                    this.logger.LogTrace("(-)[NO_BROADCAST_ITEMS]");
-                    return;
-                }
-
-                IReadOnlyNetworkPeerCollection nodes = this.connection.ConnectedNodes;
-                if (!nodes.Any())
-                {
-                    this.logger.LogTrace("(-)[NO_NODES]");
-                    return;
-                }
+                    IReadOnlyNetworkPeerCollection nodes = this.connection.ConnectedNodes;
+                    if (!nodes.Any())
+                    {
+                        this.logger.LogTrace("(-)[NO_NODES]");
+                        continue;
+                    }
 
-                // Announce the blocks to each of the peers.
-                IEnumerable<BlockStoreBehavior> behaviours = nodes.Select(s => s.Behavior<BlockStoreBehavior>());
+                    // Announce the blocks to each of the peers.
+                    IEnumerable<BlockStoreBehavior> behaviours = nodes.Select(s => s.Behavior<BlockStoreBehavior>());
 
-                this.logger.LogTrace("{0} blocks will be sent to {1} peers.", broadcastItems.Count, behaviours.Count());
-                foreach (BlockStoreBehavior behaviour in behaviours)
-                    await behaviour.AnnounceBlocksAsync(broadcastItems).ConfigureAwait(false);
-            },
-            this.nodeLifetime.ApplicationStopping,
-            repeatEvery: TimeSpans.Second,
-            startAfter: TimeSpans.FiveSeconds);
+                    this.logger.LogTrace("{0} blocks will be sent to {1} peers.", broadcastItems.Count, behaviours.Count());
+                    foreach (BlockStoreBehavior behaviour in behaviours)
+                        await behaviour.AnnounceBlocksAsync(broadcastItems).ConfigureAwait(false);
+                }
```

Comment:
here should be the exit log, but we will have no such delegate anyway, so just for future ....

Reply:
Fixed.